geode-lucene/src/main/java/org/apache/geode/cache/lucene/LuceneService.java

```diff
@@ -18,53 +18,77 @@
- * which is a singleton at each JVM. 
- * 
- * At each server JVM, for data region to create index, create the index on fields with default analyzer:
- * LuceneIndex index = luceneService.createIndex(indexName, regionName, "field1", "field2", "field3"); 
- * or create index on fields with specified analyzer:
+ * {@code
+ * LuceneIndex index = luceneService.createIndex(indexName, regionName, "field1", "field2", "field3");
+ * }
+ * </pre>
+ * <p>
+ * You can also specify what {@link Analyzer} to use for each field.
+ * </p>
+ * <pre>
+ * {@code
  * LuceneIndex index = luceneService.createIndex(indexName, regionName, analyzerPerField);
- * 
- * We can also create index via cache.xml or gfsh.
- * 
- * At client side, create query and run the search:
- * 
- * LuceneQuery query = luceneService.createLuceneQueryFactory().setLimit(200).setPageSize(20)
- * .setResultTypes(SCORE, VALUE, KEY).setFieldProjection("field1", "field2")
- * .create(indexName, regionName, querystring, analyzer);
- * 
- * The querystring is using lucene's queryparser syntax, such as "field1:zhou* AND field2:gzhou@pivotal.io"
- *  
- * PageableLuceneQueryResults results = query.search();
- * 
- * If pagination is not specified:
- * List list = results.getNextPage(); // return all results in one getNextPage() call
- * or if paging is specified:
- * if (results.hasNextPage()) {
- *   List page = results.nextPage(); // return resules page by page
  * }
- * 
- * The item of the list is either the domain object or instance of {@link LuceneResultStruct}
  * </pre>
- * 
  *
+ * Indexes should be created on all peers that host the region being indexed. Clients do not need
+ * to define the index, they can directly execute queries using this service.
+ *
+ * <p>
+ * Queries on this service can return either the region keys, values, or both that match
+ * a lucene query expression. To execute a query, start with the {@link #createLuceneQueryFactory()} method.
+ * </p>
+ *
+ * <pre>
+ * {@code
+ * LuceneQuery query = luceneService.createLuceneQueryFactory()
+ *    .setLimit(200)
+ *    .create(indexName, regionName, "name:John AND zipcode:97006", defaultField);
+ * Collection<Object> results = query.findValues();
+ * }
+ * </pre>
+ *
+ * <p>
+ * The lucene index data is colocated with the region that is indexed. This means
+ * that the index data will partitioned, copied, or persisted using the same configuration
+ * options you provide for the region that is indexed. Queries will automatically be distributed
+ * in parallel to cover all partitions of a partitioned region.
+ * </p>
+ * <p>
+ * Indexes are maintained asynchronously, so changes to regions may not be immediately reflected
+ * in the index. This means that queries executed using this service may return stale results. Use
+ * the {@link #waitUntilFlushed(String, String, long, TimeUnit)} method if you need to guarantee that
+ * your query will see the latest changes, however this method should be used sparingly because it
```

Comment:
We don't actually guarantee the latest changes, just guarantee the latests changes up to the point where waitUntilFlushed is called...not sure how to explain that though...just wanted to avoid any confusion from the user where they may think all events after that flush would also be reflected.